### PR TITLE
New delegator restriction settings

### LINKED ISSUE #2348: New settings for restricting delegator rights
Make backend changes for OpenSlides/openslides-client#3477

We need new settings for the proposed functionality, I propose the following names and functions:
- `participants_forbid_delegator_in_list_of_speakers`: Users who have delegated their votes in this meeting may not add themselves to the meetings LoS's
- `participants_forbid_delegator_as_submitter`: Users who have delegated their votes in this meeting may not create motions
- `participants_forbid_delegator_as_supporter`: Users who have delegated their votes in this meeting may not support motions

All of these will be off by default to preserve current functionality, so there's no migration necessary.

Predictably this will require changes in `motion.create`, `motion.set_support_self`, `speaker.create`, `speaker.delete` and `meeting.update`, so please don't forget to update the documentation

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #2348

### CODE REVIEW

File: openslides_backend/action/actions/motion/set_support_self.py [hunk 1]
```diff
@@ -28,6 +33,16 @@ class MotionSetSupportSelfAction(MeetingUserHelperMixin, UpdateAction):
     permission_id = "motion_id"
     permission = Permissions.Motion.CAN_SUPPORT
 
+    def check_permissions(self, instance: dict[str, Any]) -> None:
+        if not len(
```
Comment (jsangmeister) [inline openslides_backend/action/actions/motion/set_support_self.py:37]: The `if` clause is redundant, right? The helper method will throw an exception if the delegator restriction applies.

File: openslides_backend/action/actions/motion/create.py [hunk 2]
```diff
@@ -154,7 +156,12 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
             forbidden_fields.add("attachment_ids")
 
         perm = Permissions.Motion.CAN_MANAGE
-        if not has_perm(self.datastore, self.user_id, perm, instance["meeting_id"]):
+        if (
+            self.check_perm_and_delegator_restriction(
+                perm, "users_forbid_delegator_as_submitter", [instance["meeting_id"]]
+            )
+            == []
+        ):
```
Comment (jsangmeister) [inline openslides_backend/action/actions/motion/create.py:164]: same here

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/motion/create.py (hunk 2 above)
Comment (luisa-beerboom) [reply openslides_backend/action/actions/motion/create.py:164]: It's necessary to retain old functionality

File: openslides_backend/action/actions/motion/set_support_self.py (hunk 1 above)
Comment (luisa-beerboom) [reply openslides_backend/action/actions/motion/set_support_self.py:37]: It's necessary to retain old functionality

File: openslides_backend/action/actions/user/delegation_based_restriction_mixin.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                        *[
+                            FilterOperator("meeting_id", "=", meeting_id)
+                            for meeting_id in meeting_ids
+                        ]
+                    ),
+                    Or(
+                        *[
+                            FilterOperator("id", "=", meeting_user_id)
+                            for meeting_user_id in operator["meeting_user_ids"]
+                        ]
```
[the remark this answers is not in the document]
Comment (jsangmeister) [reply openslides_backend/action/actions/user/delegation_based_restriction_mixin.py:79]:
```
Or(FilterOperator("id", "=", meeting_user_id) for meeting_user_id in operator["meeting_user_ids"])
```
should work

**Review by r-peschke — APPROVED** [review]

**Pull request merged**